src/datachain/toolkit/split.py

```diff
@@ -58,14 +69,16 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
 
     weights_normalized = [weight / sum(weights) for weight in weights]
 
-    resolution = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
+    rand_col = C("sys.rand")
+    if seed is not None:
+        uniform_seed = random.Random(seed).randrange(1, RESOLUTION + 1)  # noqa: S311
+        rand_col = (rand_col % (RESOLUTION + 1)) * uniform_seed  # type: ignore[assignment]
+    rand_col = rand_col % RESOLUTION  # type: ignore[assignment]
```

Comment:
@0x2b3bfa0 wait. `RESOLUTION` Should NOT be power of 2 for modulo. It should be a prime number for even distribution. `RESOLUTION = 2^31-1` is a prime (Mersenne prime to be exact), and `RESOLUTION + 1` is a power of 2, so it is much better to use `RESOLUTION` for modulo.

Since max reminder will be `RESOLUTION - 1`, we should use `RESOLUTION + 1` for multiplication below.

Reply:
1. Unless proven otherwise, we don't need `RESOLUTION` to be prime; this is just multiplication of big numbers[^1] _modulo_ **anything**.
2. If you use `2**31 - 1` for _modulo_ you should be using `2**31 - 2` for multiplication, not `2**31 + 2` lest leaving some of the items unreachable.

[^1]: More specifically two big, pseudo-random numbers: a constant `seed` and a variable a sys.rand`

Reply:
🤔

```
In [1]: import math

In [2]: RESOLUTION = 2**31 - 1

In [3]: math.log((RESOLUTION - 1) * (RESOLUTION + 1), 2)
Out[3]: 61.99999999865639
```

Reply:
```python 
>>> one = 2**31 * 2**31
4611686018427387904
>>> two = (2**31 - 1) * (2**31 + 1)
4611686018427387903
>>> one - two
1
```

Reply:
The best conversation ever 🤣
```
In [1]: (2**31 - 1) * (2**31 + 1) == 2**62 - 1
Out[1]: True
```

Reply:
Argh, of course, you're right, should be `RESOLUTION - 1` in multiplication below.